packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1239,7 +1250,7 @@ class HotRunner extends ResidentRunner {
   Future<void> cleanupAfterSignal() async {
     await stopEchoingDeviceLog();
     await hotRunnerConfig!.runPreShutdownOperations();
-    if (_didAttach) {
+    if (stopAppDuringCleanup) {
```

Comment:
Just noticed that `stopAppDuringCleanup` is inverted.

It should be set to `true` only in the `flutter run` case, and if it is `true` then the cleanup should call `exitApp`.